library/src/image.cxx

```diff
@@ -343,11 +391,15 @@ std::vector<unsigned char> image::saveBuffer(SaveFormat format) const
   switch (format)
   {
     case SaveFormat::PNG:
-      return this->Internals->SaveBuffer<vtkPNGWriter>();
+    {
+      vtkSmartPointer<vtkPNGWriter> writer = vtkSmartPointer<vtkPNGWriter>::New();
+      this->Internals->writePngMetadata(writer);
+      return this->Internals->SaveBuffer<vtkPNGWriter>(writer);
+    }
     case SaveFormat::JPG:
-      return this->Internals->SaveBuffer<vtkJPEGWriter>();
+      return this->Internals->SaveBuffer<vtkJPEGWriter>(vtkSmartPointer<vtkJPEGWriter>::New());
```

Comment:
I think the compiler can deduce the type now

```suggestion
      return this->Internals->SaveBuffer(vtkSmartPointer<vtkJPEGWriter>::New());
```

Reply:
I think I tried but it complained, I'll double check

Reply:
it can infer if we change the function prototype to use a smart pointer
```diff
-  std::vector<unsigned char> SaveBuffer(WriterType* writer)
+  std::vector<unsigned char> SaveBuffer(vtkSmartPointer<WriterType> writer)
```
would that be good?